### PR TITLE
Assigning Buses to Drivers

### PR DESCRIPTION
**Pull request by 41hulk**
### What does this PR do?
• Assign Driver to Buses
### Description of Task to be completed?
• Appropriate error messages are displayed on unsuccessful attempts
• Authenticated users are able to assign bus to drivers.
• Skeleton UIs/screen are/is shown when the page loads/is loading.
• The app layouts are displayed
• The page is responsive.
### How should this be manually tested?
• checkout this branch
• run npm install
•  click on a row 
• click add bus to a Driver
### What are the relevant trello id?
bD45iW2U

### CODE REVIEW

**Review by GuevaraMarius — DISMISSED** [review]

**Review by GuevaraMarius — DISMISSED** [review]

**Review by mansurissa — DISMISSED** [review]
LGTM!

File: src/components/AssignBusToDriver/assignDriver.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                    value={dropDownAssign} data-testid="selectmenu">
+   <option value="" disabled selected>{t('assignDriver.choosePlate')}</option>          
+   { getBusState.data && getBusState.data.map((data)=>(
+    <option value={data.id} data-testid="option">{data.plateNo}</option>
+    ))}
+    {!getBusState.data && <option>loading</option>}
+      
+  </select>
+                </div>
+                {/*footer*/}
```
Comment (raymond42) [inline src/components/AssignBusToDriver/assignDriver.js]: remove commented line

File: src/components/AssignBusToDriver/assignDriver.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                 
+                </div> 
+                {/*body*/}
+                <div className="relative p-6 flex-auto">
+                <span class="px-1 text-sm text-gray-600">PlateNo</span>
+                {/* <input placeholder="" 
+                       type="text"
+                       class="text-md block px-3 py-2 rounded-lg w-full bg-white border-2 border-gray-300 placeholder-gray-600 shadow-md focus:placeholder-gray-500 focus:bg-white focus:border-gray-600 focus:outline-none">
+                </input>
+                <label for="cars">Choose a car:</label> */}
```
Comment (raymond42) [inline src/components/AssignBusToDriver/assignDriver.js]: remove these commented lines

File: src/components/AssignBusToDriver/assignDriver.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      { state && state.map((data) => (
+                  <tr key={data.id} className="flex w-full mb-3 hover:bg-orange-100"
+                  style={{ transition: "all .15s ease" }}
+                  onClick={() => {setShowModal(true);
+                                  setClickedDriverId(data.id)
+                          }}
+                  data-testid="assignData">
+
+                    <td className='p-4 w-1/4'>
+                      {data.firstName}
```
Comment (raymond42) [inline src/components/AssignBusToDriver/assignDriver.js]: let' use es6 destructuring to get firstName, lastName and the busId . i.e `const {firstName, LastName, budId} = data`

File: src/redux/actionCreators/assignAction.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    
+} from '../actionTypes/actionTypes';
+import {token} from '../../components/AssignBusToDriver/token';
+
+export const getDriversAction = () => async(dispatch) => {
+    
+    
+    try {
+        const response = await axios
+        .get('https://incredible-phantom.herokuapp.com/api/users',{
```
Comment (raymond42) [inline src/redux/actionCreators/assignAction.js]: we stored this url in a constant called `BACKEND_URL` you can import it and use it

File: src/tests/redux/actionCreators/assignDriverAction.test.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+              success: false,
+              message: 'error',
+              data: {}
+            }
+          });
+        });
+        await store.dispatch(getBusAction());
+        await flushPromises();
+        const calledActions = store.getActions();
+        // expect(calledActions[1].type).toEqual(GET_BUS_FAILED);
```
Comment (raymond42) [inline src/tests/redux/actionCreators/assignDriverAction.test.js]: remove this commented line

**Review by raymond42 — CHANGES_REQUESTED** [review]
Good work at @41hulk . Only a few comments to attend to

**Comment by raymond42** [comment]
and also you need to squash your commits into one commit

**Review by raymond42 — APPROVED** [review]

**Pull request merged**